naga/src/front/wgsl/error.rs

```diff
@@ -1043,6 +1056,50 @@ impl<'a> Error<'a> {
                     .into()],
                 }
             }
+            Error::DiagnosticAttributeNotYetImplementedAtParseSite {
+                site_name_plural,
+                ref spans,
+            } => ParseError {
+                message: "`@diagnostic(…)` attribute(s) not yet implemented".into(),
+                labels: {
+                    let mut spans = spans.iter().cloned();
+                    let first = spans
+                        .next()
+                        .map(|span| {
+                            (
+                                span,
+                                format!("can't use this for {site_name_plural} (yet)").into(),
+                            )
+                        })
+                        .expect("internal error: diag. attr. rejection on empty map");
+                    std::iter::once(first)
+                        .chain(spans.map(|span| (span, "".into())))
+                        .collect()
+                },
+                notes: vec![format!(concat!(
+                    "Let Naga maintainers know that you ran into this at ",
+                    "<https://github.com/gfx-rs/wgpu/issues/5320>, ",
+                    "so they can prioritize it!"
+                ))],
+            },
+            Error::DiagnosticAttributeNotSupported {
+                on_what_plural,
+                ref spans,
+            } => ParseError {
+                message: format!(
+                    "`@diagnostic(…)` attribute(s) {on_what_plural}, which are not supported"
+                ),
+                labels: spans
+                    .iter()
+                    .cloned()
+                    .map(|span| (span, "".into()))
+                    .collect(),
+                notes: vec![
+                    "`@diagnostic(…)` attributes are only permitted on `fn`s and statements."
```

Comment:
```suggestion
                    "`@diagnostic(…)` attributes are only permitted on `fn`s, some statements and switch/loop body."
```

Do we want to list supported statement kinds?

Reply:
Just force-pshed e5120a3ae98dacabc2d54d1e27af293fe687888d incorporating your suggestion, with some tweaks to noun plurality (s/body/bodies), punctuation (added an Oxford comma), and styling (backticks).